### PR TITLE
Homo lumo

### PR DESCRIPTION
**Pull request by baritone-tut0211**
## Description
Show Dipole moment and center of charge
and little change of HOMO LUMO energy

## Changelog description
Show Dipole moment and center of charge
and little change of HOMO LUMO energy

## Status

- [ ] Code base linted
- [ ] Ready to go

### CODE REVIEW

File: qcengine/programs/nwchem/harvester.py [hunk 1]
```diff
@@ -614,10 +615,50 @@ def harvest_outfile_pass(outtext):
...
+        psivar["CENTER OF CHARGE"] = np.array([mobj.group(1), mobj.group(2), mobj.group(3)])
+    
+    mobj = re.search(
+        r"Dipole moment" + r".*?" + r"A\.U\." + r"\s+" + r"DMX\s+([+-]?\d+[.]\d+)\s+" + r"DMXEFC\s+[+-]?\d+[.]\d+\s+" + 
+        r"DMY\s+([+-]?\d+[.]\d+)\s+"  + r"DMYEFC\s+[+-]?\d+[.]\d+\s+" + r"DMZ\s+([+-]?\d+[.]\d+)\s+" +
+        r"DMZEFC\s+[+-]?\d+[.]\d+\s+" + r"\-EFC\-" + r".*?" + r"A\.U\.\s+" + r"Total dipole\s+([+-]?\d+[.]\d+\s+)", outtext, re.MULTILINE)
+    # + r"DMY\s+" + r"([+-]?\d+[.]\d+)", outtext, re.MULTILINE)
+    if mobj:
+        psivar["DIPOLE MOMENT"] = np.array([mobj.group(1), mobj.group(2), mobj.group(3)])
+        psivar["TOTAL DIPOLE MOMENT"] = mobj.group(4)
```
Comment (dgasmith) [inline qcengine/programs/nwchem/harvester.py]: Cool! Can you add tests for these psivars to make sure we are parsing them correctly?

File: qcengine/programs/nwchem/harvester.py [hunk 2]
```diff
@@ -599,7 +600,7 @@ def harvest_outfile_pass(outtext):
             # print (mobj.group(2)) #error reason
             psivar['NWCHEM ERROR CODE'] = mobj.group(1)
             # TODO process errors into error var
-
+
```
Comment (loriab) [inline qcengine/programs/nwchem/harvester.py]: be sure to run `make format` as the PR box describes to get rid of little things like this. (ping if that doesn't make sense.)

File: qcengine/programs/nwchem/harvester.py [hunk 3]
```diff
@@ -614,10 +615,50 @@ def harvest_outfile_pass(outtext):
...
+                psivar["LUMO"] = round(lumo, 10)
+    
     mobj = re.search(r"AO basis - number of functions:\s+(\d+)\s+number of shells:\s+(\d+)", outtext, re.MULTILINE)
     if mobj:
         psivar["N MO"] = mobj.group(2)
         psivar["N BASIS"] = mobj.group(1)
+    
+    #Search for Center of charge
+    mobj = re.search(
+        r"Center of charge \(in au\) is the expansion point" + r"\n" + r"\s+" + r"X\s+=\s+([+-]?\d+[.]\d+)" + r"\s+" +
```
Comment (loriab) [inline qcengine/programs/nwchem/harvester.py]: could you use `NUMBER` here to simplify some regexes?

File: qcengine/programs/nwchem/harvester.py [hunk 4]
```diff
@@ -614,10 +615,50 @@ def harvest_outfile_pass(outtext):
...
     if mobj:
         psivar["N MO"] = mobj.group(2)
         psivar["N BASIS"] = mobj.group(1)
+    
+    #Search for Center of charge
+    mobj = re.search(
+        r"Center of charge \(in au\) is the expansion point" + r"\n" + r"\s+" + r"X\s+=\s+([+-]?\d+[.]\d+)" + r"\s+" + 
+        r"Y\s+=\s+([+-]?\d+[.]\d+)" + r"\s+" + r"Z\s+=\s+([+-]?\d+[.]\d+)", outtext, re.MULTILINE)
+    if mobj:
+        psivar["CENTER OF CHARGE"] = np.array([mobj.group(1), mobj.group(2), mobj.group(3)])
```
Comment (loriab) [inline qcengine/programs/nwchem/harvester.py]: What particular center of charge is this? Could it be calculated from molecule?

File: qcengine/programs/nwchem/harvester.py [hunk 5]
```diff
@@ -614,10 +615,50 @@ def harvest_outfile_pass(outtext):
...
+       
+
+        mobj = re.search(r"Vector" + r"\s+" + r"%d"%(psivar["N ALPHA ELECTRONS"] + 1) +  r"\s+" + r"Occ=" + r".*" + r"\s+" + r"E=" + r"([+-]?\s?\d+[.]\d+)" + r"[D]"+ r"([+-])" + r"[0]" + r"(\d+)", outtext, re.MULTILINE)
+        if mobj:
+            if mobj.group(2) == "+":
+                lumo = float(mobj.group(1)) * (10**(-1 * float(mobj.group(3))))
+                psivar["LUMO"] = round(lumo, 10)
+            else:
+                lumo = float(mobj.group(1)) * (10**(-1 * float(mobj.group(3))))
+                psivar["LUMO"] = round(lumo, 10)
```
Comment (loriab) [inline qcengine/programs/nwchem/harvester.py]: would you mind posting the text snippet you're targeting here? looks a little delicate.

**Comment by dgasmith** [comment]
Anything we can do to help this PR along?

**Pull request closed without merging**